### ISSUE TITLE
no-useless-escape could flag exact location of invalid escape sequence

### ISSUE THREAD

**Issue by platinumazure** [eslint/eslint#5751]
This is an enhancement request for `no-useless-escape`. Right now, we report on the literal node, which results in an error message location that points to the beginning of the string or RegExp; it would be awesome if we could take into account the offset of the offending character so that we could give a more precise location.

The issue is more obvious when you have multiple problems in your literal:

```js
/* eslint no-useless-escape: "error" */
var myString = "\ab\c";

// error 1:16 - invalid escape \a (ideally this would say 1:17)
// error 1:16 - invalid escape \c (ideally this would say 1:20)
```

Please let me know if I can make this more clear.

**Comment by mysticatea** [comment]
I like this idea.

I think `error 1:20 - Unexpected escape sequence: '\c'` or something like is good.
`no-unused-vars` is similar, the rule is warning at variable names in a comment node.

![image](https://cloud.githubusercontent.com/assets/1937871/14199974/e4e2f6ee-f822-11e5-8ae4-ed0b281e85c2.png)

**Comment by BYK** [comment]
I don't see a reason for not doing this. Gonna champion this and @onurtemizkan mentioned he already has a patch waiting so accepting.
